awx/main/utils/common.py

```diff
@@ -742,6 +744,37 @@ def ignore_inventory_computed_fields():
         _inventory_updates.is_updating = previous_value
 
 
+def _schedule_task_manager():
+    from awx.main.scheduler.tasks import run_task_manager
+    from django.db import connection
+    # runs right away if not in transaction
+    connection.on_commit(lambda: run_task_manager.delay())
+
+
+@contextlib.contextmanager
+def task_manager_bulk_reschedule():
+    """Context manager to avoid submitting task multiple times.
+    """
+    try:
+        previous_flag = getattr(_task_manager, 'bulk_reschedule', False)
+        previous_value = getattr(_task_manager, 'needs_scheduling', False)
+        _task_manager.bulk_reschedule = True
+        _task_manager.needs_scheduling = False
+        yield
+    finally:
+        _task_manager.bulk_reschedule = previous_flag
+        if _task_manager.needs_scheduling:
+            _schedule_task_manager()
+        _task_manager.needs_scheduling = previous_value
```

Comment:
The `finally:` block here deals with a similar question that Django transactions deal with. If you connect something `on_commit` inside of a nested transaction, then it will _not_ run at the end of that transaction. Instead, it will delay until the outermost transaction has committed.

A similar argument might apply here. If we are gathering the task manager submissions, then an inner loop could defer to an outer loop. In that case, if `previous_flag` was true, then this would set `needs_scheduling` to True here, and then be done.

I did _not_ do that, because it just seemed like over-thinking it when the actual use case is way more simple.